src/large_pages/node_large_page.cc

```diff
@@ -282,6 +282,25 @@ bool IsSuperPagesEnabled() {
 }
 #endif
 
+class MmapPointer {
+ public:
+  void
+  DoMmap(void* start, size_t size_, int prot, int flags, int fd, int offset) {
+    mem = mmap(start, size_, prot, flags, fd, offset);
+    size = size_;
+  }
+  bool operator==(void* rhs) { return mem == rhs; }
+  operator void*() { return mem; }
```

Comment:
```suggestion
  bool operator==(void* rhs) const { return mem == rhs; }
  operator void*() const { return mem; }
```

Reply:
A bool operator in here would be good also... to allow for...

```c++
MmapPointer foo(/*...*/);
// ...
if (!foo) { /* handle */ }
```

Reply:
IINM we're discouraged by the linter from treating pointers as booleans. IIRC we're always supposed to compare to `nullptr`.

Reply:
Smart pointers are a bit of an exception to that. We have several `operator bool() const` in the code allowing for the simplified check. *Those* do compare to `nullptr` internally so that the pointer is not treated as a boolean.

Reply:
Also, we don't compare to `nullptr` in the code consuming `MemoryMapPointer`, but rather to MAP_FAILED, which is a `void*`. I can add the operator, but it would be unused.

Reply:
We've moved in a slightly different direction with @bnoordhuis' input. We now compare pointers explicitly outside the class.